### PR TITLE
Feature: Signature verification for Solana and Ethereum messages

### PR DESCRIPTION
**Pull request by MHHukiewitz**
Problem: Users cannot verify signatures

Solution: Implement verify_signature() functions for **Ethereum** and **Solana** signatures, more are following.

Partly solves: #9

### CODE REVIEW

File: src/aleph/sdk/chains/ethereum.py [hunk 1]
```diff
@@ -41,3 +41,39 @@ def get_public_key(self) -> str:
...
+    if isinstance(message, bytes):
+        message = message.decode("utf-8")
+
+    message_hash = encode_defunct(text=message)
+    try:
+        address = Account.recover_message(message_hash, signature=signature)
+        if address == public_key:
+            return True
+        else:
+            return False
```
Comment (BjrInt) [inline src/aleph/sdk/chains/ethereum.py:77]: Could this ever happen though ?

Comment (MHHukiewitz) [reply src/aleph/sdk/chains/ethereum.py:77]: It's just the way the recover_message function works \o/

**Review by BjrInt — APPROVED** [review]
Looks good to me. :+1:

**Comment by hoh** [comment]
Related to https://github.com/aleph-im/aleph-sdk-python/issues/9

**Pull request merged**